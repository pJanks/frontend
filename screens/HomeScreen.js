import * as React from 'react';
import { View, Button, Text, StyleSheet } from 'react-native';
import UserWelcome from '../components/UserWelcome';
import WeatherWindow from '../components/WeatherWindow';
import UserLandingPage from '../components/UserLandingPage';

export default function HomeScreen({ navigation }) {
  return (
    <View>
      <UserWelcome />
      <WeatherWindow />
      <View style={ styles.createActivityButton }>
        <Button
          buttonStyle={ styles.createActivityButton }
          title='Create New Activity'
          onPress={() => navigation.navigate('New Activity')} />
      </View>
      <Text>userActivities</Text>
      <UserLandingPage />
    </View>
  )
}

const styles = StyleSheet.create({
  createActivityButton: {
    margin: 20,
  }
})
