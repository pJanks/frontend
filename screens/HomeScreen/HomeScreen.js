import React, { useState, useEffect } from 'react';
import { View, Button, StyleSheet, Text } from 'react-native';
import UserWelcome from '../../components/UserWelcome/UserWelcome';
import UserLandingPage from '../../components/UserLandingPage/UserLandingPage';
import { fetchAllUserActivities } from '../../apiCalls/apiCalls';
import Loading from '../../components/Loading/Loading';

export default function HomeScreen({ navigation }) {
  const [activities, setActivities] = useState([]);
  const [userInfo, setUserInfo] = useState()

  useEffect(() => {
    if (!activities.length) {
      fetchAllUserActivities(1)
        .then(data => {
          setUserInfo({id: data.id, firstName: data.first_name, lastName: data.last_name})
          setActivities(data.scheduled_activities)
        })
    }
  })
  if (activities && userInfo) {
    return (
      <View>
        <UserWelcome userInfo={userInfo} />
        <View style={ styles.createActivityButton }>
          <Button
            buttonStyle={ styles.createActivityButton }
            title='Create New Activity'
            onPress={() => navigation.navigate('NewActivity', {userInfo})} />
        </View>
        <UserLandingPage activities={ activities } userInfo={userInfo} />
      </View>
    )
  } else {
    return <Loading />
  }
}

const styles = StyleSheet.create({
  createActivityButton: {
    margin: 20,
  },
  // activityHeader: {
  //   fontSize: 20
  // }
})
