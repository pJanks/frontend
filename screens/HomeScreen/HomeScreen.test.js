import React from 'react';
import HomeScreen from './HomeScreen';

import renderer from 'react-test-renderer';

test('testing suite runs', () => {
  expect(true).toEqual(true)
})

// test('renders correctly', () => {
//
//   const HomeScreen = renderer.create(<HomeScreen />).toJSON();
//
//   expect(HomeScreen).toMatchSnapshot();
// });
