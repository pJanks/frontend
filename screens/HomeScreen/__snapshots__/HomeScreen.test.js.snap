// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should load with props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#b2e1f4",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 10,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#07407b"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
